mock_analytics_provider.py

```diff
@@ -0,0 +1,18 @@
+class MockAnalyticsProvider(object):
+    """A mock analytics provider that keeps track of how many times it's called."""
+
+    @classmethod
+    def from_config(cls, config):        
+        return cls(config.get('option'))
+
+    def __init__(self, option=None):
+        self.option = option
+        self.count = 0
+        self.event = None
+
+    def collect_event(self, _db, lp, event_type, time, **kwargs):
```

Comment:
This is fine for now, but if testing gets more complicated you might like to store the events in a list rather thanjust store a count and the type of the most recent event.